extensions/amp-audioburst-audio/0.1/amp-audioburst-audio.js

```diff
@@ -0,0 +1,460 @@
+    this.createSlider_();
+
+    /* Times */
+    const timeWrapper = this.win.document.createElement('div');
+    setStyles(timeWrapper, {
+      'text-align': 'center',
+      'color': 'rgba(0, 0, 0, 0.3)',
+      'margin': '0.5em auto',
+      'font-size': '0.8em',
+    });
+    this.currentTimeEl_ = this.win.document.createElement('span');
+    this.durationEl_ = this.win.document.createElement('span');
+    timeWrapper.appendChild(this.currentTimeEl_);
+    timeWrapper.appendChild(this.durationEl_);
+
+    if (this.fullShowSrc_ && this.fullShowSrc_.length > 0) {
+      this.container_.appendChild(fullShowButton);
+    }
+    this.container_.appendChild(this.slider_);
+    this.container_.appendChild(timeWrapper);
+    this.element.appendChild(this.container_);
+    return this.loadPromise(this.container_);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.FIXED_HEIGHT;
+  }
+
+  /** */
+  play() {
+    if (this.audio_ && !this.isPlaying) {
+      this.audio_.play();
+      this.setPlaying_(true);
+    }
+  }
+
+  /** */
+  pause() {
+    if (this.audio_ && this.isPlaying) {
+      this.audio_.pause();
+      this.setPlaying_(false);
+    }
+  }
+
+  /** @private
+   * @param {boolean} playing
+  */
+  setPlaying_(playing) {
+    if (!!playing) {
+      this.isPlaying = true;
+      if (this.playBtn_ && this.pauseBtn_) {
+        setStyle(this.playBtn_, 'display', 'none');
+        setStyle(this.pauseBtn_, 'display', 'flex');
+      }
+    } else {
+      this.isPlaying = false;
+      setStyle(this.pauseBtn_, 'display', 'none');
+      setStyle(this.playBtn_, 'display', 'flex');
+    }
+  }
+
+  /** @private */
+  timeUpdate_() {
+    if (this.audio_) {
+      this.currentTime = this.audio_.currentTime;
+      if (this.currentTimeEl_) {
+        this.currentTimeEl_.textContent = this.convertTime_(this.currentTime);
+        this.currentTimeEl_.textContent += ' / ';
+        this.updateSlider_();
+      }
+    }
+  }
+
+  /** @private */
+  setDuration_() {
+    if (this.audio_) {
+      this.duration = this.audio_.duration;
+      if (this.durationEl_) {
+        this.durationEl_.textContent = this.convertTime_(this.duration);
+      }
+    }
+  }
+
+  /** @private */
+  createPauseButton_() {
+    const btn = this.win.document.createElement('button');
+    setStyles(btn, {
+      'width': '100px',
+      'height': '100px',
+      'background': 'none',
+      'border': 'none',
+      'display': 'none',
+      'justify-content': 'center',
+      'align-items': 'center',
+      'outline': 'none',
+      'cursor': 'pointer',
+      'z-index': '10',
+    });
+    const pauseIcon = this.win.document.createElement('div');
+    setStyles(pauseIcon, {
+      'width': '30px',
+      'height': '22px',
+    });
+    const pauseIconVertLine1 = this.win.document.createElement('div');
+    const pauseIconVertLine2 = this.win.document.createElement('div');
+    const pauseIconStyles = {
+      'width': '20%',
```

Comment:
You should make a css file and add `i-amphtml-audioburst-blah` classes to the elements instead of using inline styles here.

Reply:
I tried and spent almost day on it but couldn't make it work. Anyway, it's doesn't really make sense creation of css classes, because most of elements have their unique styles and nothing gonna change. Furthermore, I'm afraid of possible class names overlap in future, so when I use inline styles it's much safer.